packages/analytics-js/src/components/userSessionManager/UserSessionManager.ts

```diff
@@ -62,52 +62,80 @@ class UserSessionManager implements IUserSessionManager {
      * Update userId in storage automatically when userId is updated in state
      */
     effect(() => {
-      this.storage?.set(userSessionStorageKeys.userId, state.session.userId.value);
+      if (state.session.userId.value === undefined) {
```

Comment:
Let's handle `undefined`, `null, and empty string values

Reply:
user can set empty string and null as userId

Reply:
Addressed as per discussion